iOS Client/PodcastsController/PodcastsCoordinator.swift

```diff
@@ -9,9 +9,14 @@
 import Foundation
 import UIKit
 import ReSwift
+import AVKit
 
-class PodcastsCoordinator: Coordinator, StoreSubscriber {
+class PodcastsCoordinator: NSObject, Coordinator, StoreSubscriber, AVAudioPlayerDelegate, PodcastHandler {
+    
     var childCoordinators: [Coordinator] = []
+    weak var playerDelegate: PodcastPlayerDelegate? {
```

Comment:
this shouldn't exist.

The point of delegation is to create a relationship at runtime between an object that has something to report and another object willing to listen by conforming to some interface.

It makes sense to make the coordinator a delegate of the view controller, where the coordinator is stored as a weak property on the VC to avoid a retain cycle. This way the VC can report events back up to the coordinator. 

However, delegation the other way doesn't make as much sense b/c the coordinator already knows about this view controller. The coordinator should just be able to directly invoke some method on it.

Reply:
Yeah I get what you're saying, the thing is that it's the view controller that needs to listen to messages from the coordinator (specifically the updating of the timestamp on the player). I would normally send this via an action and have the vc get a newState, but didn't want to flood the whole app with an action every second while the a podcast is playing.

I think your last point about the coordinator already knowing about the vc is what I can fix. I don't need another protocol, I can just call the method on the vc directly. I'll try that.

Reply:
👍